pkg/serverless/trace/inferredspan/inferred_span.go

```diff
@@ -171,5 +171,8 @@ func IsInferredSpansEnabled() bool {
 // AddTagToInferredSpan is used to add new tags to the inferred span in
 // inferredSpan.Span.Meta[]. Should be used before completing an inferred span.
 func (inferredSpan *InferredSpan) AddTagToInferredSpan(key string, value string) {
+	if inferredSpan.Span.Meta == nil {
+		inferredSpan.Span.Meta = make(map[string]string)
+	}
```

Comment:
I was getting a panic when running this test locally

Reply:
Does it make sense to move this earlier on where we create the inferred span?

Reply:
@IvanTopolcic I ran through my test with a debugger and tried putting this nil check in other spots such as `generateInferredSpan` https://github.com/DataDog/datadog-agent/blob/8c3be8a7aed4d855605a7a2f70b6b25f355f8063/pkg/serverless/trace/inferredspan/inferred_span.go#L156

but it seems my test is hitting some other code path, as we make an[ inferredspan slice ](https://github.com/DataDog/datadog-agent/pull/16156/files#diff-96b011b6370ccc9a3d01c42f36ee2bdc78f78b0144aaf5f30b6dae884021b62bR224) manually in the test then call OnInvokeEnd and thus I think it is at a good spot for the nil check.